src/main/java/com/gradle/enterprise/api/ApiProblemParser.java

```diff
@@ -5,22 +5,24 @@
 import com.gradle.enterprise.api.client.ApiException;
 import com.gradle.enterprise.api.model.ApiProblem;
 
+import java.io.UncheckedIOException;
 import java.util.Optional;
 
 public final class ApiProblemParser {
 
-    private final static ObjectMapper OBJECT_MAPPER = new ObjectMapper();
     private static final String CONTENT_TYPE = "application/problem+json";
 
-    public static Optional<ApiProblem> maybeParse(ApiException apiException) {
+    public static Optional<ApiProblem> maybeParse(final ApiException apiException, final ObjectMapper objectMapper) {
```

Comment:
Fully agree with what you said, however I put them there exactly for that reason - if one sees a parameter or variable not marked as "final", its a signal right away that somewhere in that method there is a mutation happening and its done for a reason (same with "final" classes or "final" attributes in the class, those could also be left out for the same reason you've mentioned). So flow wise, one can immediately identify while going throw the logic which things will be mutated and which not (as opposed to not knowing that in advance and having to check the algorithm in detail)

I will remove those for this PR, so that convention is adhered to and we can have this discussion separately.